internal/controllers/app_controller.go

```diff
@@ -227,6 +247,34 @@ func (r *AppReconciler) reconcile(ctx context.Context, app *ketchv1.App) reconci
 	}
 }
 
+// canaryPodStatus checks whether all pods for canary deployment are running or not.
+func canaryPodStatus(c client.Client, app *ketchv1.App) error {
+	if len(app.Spec.Deployments) <= 1 {
+		return errors.New("no canary deployment found")
+	}
+
+	// podList contains list of Pods matching the specifed labels below
+	podList := &v1.PodList{}
+	listOpts := []client.ListOption{
+		// The specified labels below matches with the 2nd deployment or Canary deployment of the app.
+		client.MatchingLabels(map[string]string{
+			"theketch.io/app-name":               app.Name,
+			"theketch.io/app-deployment-version": fmt.Sprint(app.Spec.Deployments[1].Version)}),
+	}
+
+	if err := c.List(context.Background(), podList, listOpts...); err != nil {
+		return err
+	}
+
+	// check if all pods are running for the deployment
+	for _, pod := range podList.Items {
+		if pod.Status.Phase != v1.PodRunning {
+			return errors.New("all pods are not running")
```

Comment:
What about moving this error, and error at the begin of function into variables?

Reply:
Since we are not reusing this error now, so it's ok to keep it here. :)